mhkit/wave/io/wind_toolkit.py

```diff
@@ -0,0 +1,251 @@
+import pandas as pd
+from rex import MultiYearWindX
+import matplotlib.pyplot as plt
+
+
+def region_selection(lat_lon, preferred_region=''):
+    '''
+    Returns the name of the predefined region in which the given coordinates reside.
+    Can be used to check if the passed lat/lon pair is within the WIND Toolkit hindcast dataset. 
+
+    Parameters
+    ----------
+    lat_lon : list or tuple
+        Latitude and longitude coordinates as floats or integers
+        
+    preferred_region : string (optional)
+        Latitude and longitude coordinates as floats or integers
+    
+    Returns
+    -------
+    region : string
+        Name of predefined region for given coordinates
+    '''
+    assert isinstance(lat_lon, (list,tuple)), 'lat_lon must be of type list or tuple'
+    assert isinstance(lat_lon[0], (float,int)), 'lat_lon values must be of type float or int'
+    assert isinstance(lat_lon[1], (float,int)), 'lat_lon values must be of type float or int'
+    assert isinstance(preferred_region, str), 'preferred_region must be of type string'
+    
+    rDict = {
+        'CA_NWP_overlap':{'lat':[41.213, 42.642], 'lon':[-129.090, -121.672]},
+        'Offshore_CA':{   'lat':[31.932, 42.642], 'lon':[-129.090, -115.806]},
+        'Hawaii':{        'lat':[15.565, 26.221], 'lon':[-164.451, -151.278]},
+        'NW_Pacific':{    'lat':[41.213, 49.579], 'lon':[-130.831, -121.672]},
+        'Mid_Atlantic':{  'lat':[37.273, 42.211], 'lon':[-76.427, -64.800]},
+    }
+
+    region_search = lambda x: all( ( True if rDict[x][dk][0] <= d <= rDict[x][dk][1] else False
+                                    for dk, d in {'lat':lat_lon[0],'lon':lat_lon[1]}.items() ) )
+    region = [key for key in rDict if region_search(key)]
+    
+    if region[0] == 'CA_NWP_overlap':
+        if preferred_region == 'Offshore_CA':
+            region[0] = 'Offshore_CA'
+        elif preferred_region == 'NW_Pacific':
+            region[0] = 'NW_Pacific'
+        else:
+            raise TypeError(f"Preferred_region ({preferred_region}) must be 'Offshore_CA' or 'NW_Pacific' when lat_lon {lat_lon} falls in the overlap region")
+        
+    if len(region)==0:
+        raise TypeError(f'Coordinates {lat_lon} out of bounds. Must be within {rDict}')
+    else:
+        return region[0]
+
+def plot_region(region,ax=None):
```

Comment:
I think we should add this to the example notebook. Seems like a cool function to highlight.

Reply:
What do you think about passing the lat_lon value instead of the region?  I think showing a lat_lon value in the plot would be useful for helping the user understand where their lat lon is within the region.

I also think plotting multiple regions could be useful, especially for the overlap region. 

![image](https://user-images.githubusercontent.com/13438942/190685942-0bcc6844-881b-4a33-9f63-bf54580a05b1.png)

I do not feel strongly about expanding this function, I just like plotting functions.

Reply:
Sure I can add this to the example notebook, it's a good idea to make sure users know it exists. 

Do you know a way to isolate the boundary points from a list? The variable is just downloaded as a massive list of all points in the region, sometimes in the range of 100,000-1,000,000 points. It would make it a lot easier to plot multiple regions and overlap with just the boundaries.

Reply:
Yes 😄 actually in #61 I posted some code that does just that although it was never implemented it would work as needed for this case. I was using an alpha shape approach. If you are interested in other approaches the top answer from [this](https://gis.stackexchange.com/questions/5426/finding-boundary-co-ordinates-from-given-set-of-point-co-ordinates) question has what appears to be a good summary of other options. 

Its always best to see if these exist in any of our current MHKiT dependencies but alpha shape did not seem to be there from quick look.

Reply:
@ssolson I updated the notebook to highlight each of the provided functions, including the plotting one. In ``plot_region()`` users can now specify a region and their specific lat-lon point to show on top of that region. I rethought the boundary idea, and implementing robust alpha shapes seems like a lot of work and potential upkeep. I left the plots as is, so they look like a solid-colored block of that region, instead of the outline. I guess that it would always make the plotting function more expense, but have not checked. But because I left that out, plotting the overlap is hard. Each region would just lie on top of the other without clear boundaries. Users could instead use ``region_selection`` to check for the overlap region, or use ``plot_region`` twice to show both the California and the NW Pacific regions. 
![image](https://user-images.githubusercontent.com/72414466/192373164-40f3e6ae-fd5d-4403-99c0-7cb8efb052c8.png)

Reply:
I think highlighting the point is pretty cool. Emily and I will be putting alpha shapes in for D3D plotting so if there is any interest in revisiting the outline at a later date we can.